Lg/dcos 50125 public ip cli (#2123)
mesosphere/dcos-docs-site: pages/1.13/administering-clusters/locate-public-agent/index.md

excerpt: Finding a public agent IP address.
enterprise: false
---
After you have installed DC/OS with a public agent node declared, you can navigate to the public IP address of your public agent node. You can expose the public-facing IP address for an agent as a gateway for access to services that are running inside the DC/OS cluster. For example, if you are configuring load balancing to distribute inbound requests to the services in a cluster, the requests are typically routed through the public IP address frontend to an appropriate service instance backend isolated behind a firewall. 

# Before you begin
- You must have DC/OS installed with at least one master node and at least one [public agent](/1.13/overview/concepts/#public-agent-node) node.
- You must have the most recent version of the DC/OS [CLI](/1.13/cli/) installed.

> - You must have the [jq](https://github.com/stedolan/jq/wiki/Installation) language installed if you want to format the output from an API call.

Reviewer: I've read too many RFC and `must` means its mandatory ;).

```suggestion
- You may use [jq](https://github.com/stedolan/jq/wiki/Installation) if you want to format the output from an API call.
```

Author: fixed.

The DC/OS application programming interface (API) provides the underlying funtionality that you can access through the DC/OS web-based administrative console and command-line interface (CLI). In most cases, therefore, you only use the API directly if you are integrating the functionality with a custom program or automation script. However, you can retrieve the public IP addresses for public agents directly through calls to the DC/OS application programming interface (API) for networking, if needed.

To find public IP addresses by using the DC/OS API:
1. Identify the URL for a DC/OS cluster running on a public cloud instance.

1. Use the following REST API `net/v1/nodes` endpoint to find the public IP of the public agents:

    ```bash
    <cluster-url>/net/v1/nodes
    ```

    For example, you might issue a call with a cluster URL similar to the following to specify where the DC/OS cluster is running:
    ```bash
    http://luxi-sf7-elasticl-u70m3un6kcab-1100943753.us-west-2.elb.amazonaws.com/net/v1/nodes
    ```

1. Review the command output.

    This API call returns information similar to the following in an unfiltered format:

    ```bash

>     [{"updated":"2019-01-07T22:22:22.171Z","public_ips":["34.212.37.79"],"private_ip":"10.0.6.210","hostname":"ip-10-0-6-210"},{"updated":"2019-01-07T22:22:22.119Z","public_ips":["52.25.254.97"],"private_ip":"10.0.6.181","hostname":"ip-10-0-6-181"},{"updated":"2019-01-07T22:21:09.585Z","public_ips":["54.218.23.75"],"private_ip":"10.0.6.148","hostname":"ip-10-0-6-148"},{"updated":"2019-01-07T22:22:28.582Z","public_ips":[],"private_ip":"10.0.1.139","hostname":"ip-10-0-1-139"},{"updated":"2019-01-07T22:22:28.649Z","public_ips":[],"private_ip":"10.0.0.138","hostname":"ip-10-0-0-138"}]

Reviewer: nit: I think formatting all the json (though `jq` or similar) would help with readability. I always tend to skip `json` blobs that arn't formatted, if I find out later that it does have info in it I need to understand, copy it to a terminal to format it. Perhaps above have a note that raw api responses are in compact form and have been formatted in the documentation for readability or something like that.

Author: good idea

    ```

>     In this example, there are three IP addresses for the public agents:

Reviewer: One of these IPs is on the master node (assuming this was from a 1 master and 2 public, 2 private agent setup). Not sure really how to phrase this correctly, but technically not all the ips are on public agents.

```bash
curl http://localhost:62080/v1/nodes
```

This command output displays the public IP addresses of the same public agents that are running on the DC/OS cluster. For example: 

    ```bash
    curl http://localhost:62080/v1/nodes
    [{"updated":"2019-01-07T22:22:22.171Z","public_ips":["34.212.37.79"],"private_ip":"10.0.6.210","hostname":"ip-10-0-6-210"},{"updated":"2019-01-07T22:22:22.119Z","public_ips":["52.25.254.97"],"private_ip":"10.0.6.181","hostname":"ip-10-0-6-181"},{"updated":"2019-01-07T22:21:09.585Z","public_ips":["54.218.23.75"],"private_ip":"10.0.6.148","hostname":"ip-10-0-6-148"},{"updated":"2019-01-07T22:22:28.582Z","public_ips":[],"private_ip":"10.0.1.139","hostname":"ip-10-0-1-139"},{"updated":"2019-01-07T22:22:28.649Z","public_ips":[],"private_ip":"10.0.0.138","hostname":"ip-10-0-0-138"}]
    ```

    In both of these examples, there are three public IP addresses, one for each of the public agent nodes:
    - 34.212.37.79
    - 52.25.254.97
    - 54.218.23.75

    As with the previous example, there are no public IP addresses available for the private nodes. 

> ## Formatting API output using jq

Reviewer: In concert with my above note, perhaps move this earlier in the doc, show both an formatted and formatted response and make the note that from here on out all responses have been formatted for readability.

Author: great suggestion.
